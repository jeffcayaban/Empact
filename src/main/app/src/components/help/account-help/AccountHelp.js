import React from 'react';
import {Col, Grid, ListGroup, ListGroupItem, Row} from "react-bootstrap";
import BackButton from "../../common/BackButton";

/**
 * Displays the links to help pages that are relevant to accounts.
 */

class AccountHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Account Help</h2>
                </div>

                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <ListGroup>
                                    <ListGroupItem onClick={() => window.location = `/help/account/profile`}>
                                        Your Profile
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/account/create`}>
                                        Creating an account
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/account/delete`}>
                                        Deleting your account
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/account/login`}>
                                        Logging in
                                    </ListGroupItem>
                                    <ListGroupItem onClick={() => window.location = `/help/account/logout`}>
                                        Logging out
                                    </ListGroupItem>
                                </ListGroup>
                                <BackButton link={"/help"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default AccountHelp;