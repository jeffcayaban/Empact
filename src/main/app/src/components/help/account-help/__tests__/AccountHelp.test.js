import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import AccountHelp from "../AccountHelp";

describe('AccountHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<AccountHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
