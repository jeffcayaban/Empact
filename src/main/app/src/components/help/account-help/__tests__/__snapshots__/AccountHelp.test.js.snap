// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Account Help
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <ListGroup
            bsClass="list-group"
          >
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Your Profile
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Creating an account
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Deleting your account
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Logging in
            </ListGroupItem>
            <ListGroupItem
              bsClass="list-group-item"
              listItem={false}
              onClick={[Function]}
            >
              Logging out
            </ListGroupItem>
          </ListGroup>
          <BackButton
            link="/help"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
